docs/sources/reference/components/prometheus.exporter.snowflake.md

```diff
@@ -20,18 +20,34 @@ prometheus.exporter.snowflake "LABEL" {
 }
 ```
 
+## Usage - RSA Authentication
+
+```alloy
+prometheus.exporter.snowflake "LABEL" {
+    account_name =         ACCOUNT_NAME
+    username =             USERNAME
+    private_key_path =     /PATH/TO/rsa_key.p8
+    private_key_password = PASSWORD
+    warehouse =            WAREHOUSE
+}
+```
+
 ## Arguments
 
 The following arguments can be used to configure the exporter's behavior.
 Omitted fields take their default values.
-
-| Name           | Type     | Description                                           | Default          | Required |
-| -------------- | -------- | ----------------------------------------------------- | ---------------- | -------- |
-| `account_name` | `string` | The account to collect metrics for.                   |                  | yes      |
-| `username`     | `string` | The username for the user used when querying metrics. |                  | yes      |
-| `password`     | `secret` | The password for the user used when querying metrics. |                  | yes      |
-| `role`         | `string` | The role to use when querying metrics.                | `"ACCOUNTADMIN"` | no       |
-| `warehouse`    | `string` | The warehouse to use when querying metrics.           |                  | yes      |
+One of `password` or `private_key_path` must be specified to authenticate.
```

Comment:
Not sure what they best way to relay this info is. Could also do something in the default section. Maybe something like 
```
yes (only if `private_key_path` not provided)
```
and vice-versa

Reply:
I added these comments above to avoid bloating the table with all the additional conditions, but maybe Grafana doesn't mind. It feels wrong to mark all 3 of password, path, and pk password as required, but there are technically scenarios when they are separately required.